### PR TITLE
Angular refactoring

### PR DESCRIPTION
**Pull request by comfi01**
## Angular refactoring

1. Shuffle lending/lib.rs structure around
2. Go through Errors by  (1) deleting unused ones and (2) renaming confusing ones
3. Move type aliases to Pallet file
4. Rename MarketIndex to MarketId
5. Rename calculate_initial_pool_size
6. Remove unneeded validity checks
7. Move all impl’s to the [lib.rs](http://lib.rs/) file
8. Remove the [impls.rs](http://impls.rs/) file
9. Move offchain worker logic to helpers/offchain_workers
10. Move borrow logic into helpers/borrow.rs
11. Move Lending trait’s repay_borrow logic + [repay_borrow.rs](http://repay_borrow.rs/) into helpers/repay_borrow.rs
12. remove [repay_borrow.rs](http://repay_borrow.rs/) file
13. Move deposit_events to trait impl
14. Add calculate_weight method to InitializeBlockCallCounters
15. Move low-level functionality to src/helpers/___ where ___ is the corresponding functionality as a file name (i.e. [borrow.rs](http://borrow.rs/))

### CODE REVIEW

File: frame/lending/src/helpers/borrow.rs [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+		Ok(())
+	}
+
+	// Checks if we can borrow from the vault.
+	// If available_funds() returns FundsAvailability::Depositable then vault is unbalanced,
+	// and we can not borrow, except the case when returned balances equals zero.
+	// In the case of FundsAvailability::MustLiquidate we obviously can not borrow, since the market
+	// is going to be closed. If FundsAvailability::Withdrawable is return, we can borrow, since
+	// vault has extra money that will be used for balancing in the next block. So, if we even
+	// borrow all assets from the market, vault has posibity for rebalancing.
```
Comment (kevin-composable) [inline frame/lending/src/helpers/borrow.rs:168]: posibity -> possibility

File: frame/lending/src/helpers/collateral.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+		let collateral_balance_after_withdrawal_value =
+			Self::get_price(market.collateral_asset, collateral_balance.safe_sub(&amount)?)?;
+
+		let borrower_after_withdrawal = BorrowerData::new(
+			collateral_balance_after_withdrawal_value,
+			borrow_balance_value,
+			market
+				.collateral_factor
+				.try_into_validated()
+				.map_err(|_| ArithmeticError::Overflow)?, // TODO: Use a proper error mesage?
```
Comment (kevin-composable) [inline frame/lending/src/helpers/collateral.rs:72]: mesage -> message

File: frame/lending/src/helpers/market.rs [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+					.collect(),
+				},
+			)?;
+
+			let initial_market_volume =
+				Self::calculate_initial_market_volume(config_input.borrow_asset())?;
+
+			ensure!(
+				initial_market_volume > T::Balance::zero(),
+				Error::<T>::IntitialMakretVolumeIncorrect
```
Comment (kevin-composable) [inline frame/lending/src/helpers/market.rs:65]: IntitialMakretVolumeIncorrect -> InitialMarketVolumeIncorrect

File: frame/lending/src/helpers/market.rs [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+				market.liquidators = input.liquidators.clone();
+				Ok(())
+			} else {
+				Err(Error::<T>::MarketDoesNotExist)
+			}
+		})?;
+		Ok(())
+	}
+
+	/// Returns pair of market's id and market (as 'MarketConfing') via market's id
```
Comment (kevin-composable) [inline frame/lending/src/helpers/market.rs:128]: MarketConfing -> MarketConfig

File: frame/lending/src/helpers/on_init.rs [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+			let mut errors = Markets::<T>::iter()
+				.map(|(market_id, config)| {
+					call_counters.read_markets += 1;
+					Self::accrue_interest(&market_id, now)?;
+					call_counters.accrue_interest += 1;
+					let market_account = Self::account_id(&market_id);
+					call_counters.account_id += 1;
+					// NOTE(hussein-aitlahcen):
+					// It would probably be more perfomant to handle theses
```
Comment (kevin-composable) [inline frame/lending/src/helpers/on_init.rs:27]: perfomant -> performant

File: frame/lending/src/helpers/on_init.rs [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+					// case while borrowing/repaying.
+					//
+					// I don't know whether we would face any issue by doing that.
+					//
+					// borrow:
+					//  - withdrawable = transfer(vault->market) + transfer(market->user)
+					//  - depositable = error(not enough borrow asset) // vault asking for reserve
+					//    to be fullfilled
```
Comment (kevin-composable) [inline frame/lending/src/helpers/on_init.rs:35]: fullfilled -> fulfilled

File: frame/lending/src/lib.rs [hunk 7]
```diff
@@ -515,6 +355,122 @@ pub mod pallet {
...
+			manager: T::AccountId,
+			currency_pair: CurrencyPair<T::MayBeAssetId>,
+		},
+		MarketUpdated {
+			market_id: MarketId,
+			input: UpdateInput<T::LiquidationStrategyId, <T as frame_system::Config>::BlockNumber>,
+		},
+		/// Event emitted when collateral is deposited.
+		CollateralDeposited { sender: T::AccountId, market_id: MarketId, amount: T::Balance },
+		/// Event emitted when collateral is withdrawed.
```
Comment (kevin-composable) [inline frame/lending/src/lib.rs:378]: withdrawed -> withdrawn

File: frame/lending/src/repay_borrow.rs [hunk 8]
```diff
@@ -9,6 +9,7 @@ use crate::Config;
 /// Release given `amount` of `debt_token` from `beneficiary_account`, transfer `amount` from
```
Comment (kevin-composable) [inline frame/lending/src/repay_borrow.rs:9]: the contents of this file should be moved to under the helpers folder

**Review by kevin-composable — COMMENTED** [review]
I had a couple small changes, but everything looked great

File: frame/lending/src/repay_borrow.rs (hunk 8 above)
Comment (comfi01) [reply frame/lending/src/repay_borrow.rs:9]: sure

**Review by kevin-composable — APPROVED** [review]

**Comment by comfi01** [comment]
bors merge

**Pull request merged**